### DISCUSSION TITLE
ran_func_id

### DISCUSSION THREAD

**Discussion by abida-02** [srsran/oran-sc-ric#47] [General]
for kpm subscription ran_func_id = 2 and for rc_xapp the ran_func_id =3. is it possible to use both the ran_func_id 2 and 3 at the same time in a single xapp?

kpm_mon_xapp is collecting ue data from ran  and printing the value as following .  My question is what is the unit of the downlink and uplink throughput value  that the xapp is printing here?
-UE_id: 3
---granulPeriod: 1000
---Metric: DRB.UEThpUl, Value: [12]
---Metric: DRB.UEThpDl, Value: [75133]
--UE_id: 1
---granulPeriod: 1000
---Metric: DRB.UEThpUl, Value: [18]
---Metric: DRB.UEThpDl, Value: [99084]
--UE_id: 2
---granulPeriod: 1000
---Metric: DRB.UEThpUl, Value: [21]
---Metric: DRB.UEThpDl, Value: [71480]
--UE_id: 3
---granulPeriod: 1000
---Metric: DRB.UEThpUl, Value: [12]
---Metric: DRB.UEThpDl, Value: [75133]
--UE_id: 1
---granulPeriod: 1000
---Metric: DRB.UEThpUl, Value: [18]
---Metric: DRB.UEThpDl, Value: [99084]

**Comment by pgawlowicz** [comment]
Here both are used in a single xApp: https://github.com/srsran/oran-sc-ric/blob/main/xApps/python/simple_xapp.py
Regarding the units, please check the relevant specifications, or also here: https://github.com/srsran/srsRAN_Project/blob/main/lib/e2/e2sm/e2sm_kpm/e2sm_kpm_metric_defs.h#L240

**Comment by abida-02** [comment]
In my setup i have enabled two slices in gnb and i connected two ue with two different slice. simple_xapp.py and kpm_mon_xapp.py both are able to collect the ue_id and dlthroughput from gnb. Is it possible to collect the slice id ( sd) , which ue is connected to the specific slice , by the xapp ?

**Reply by pgawlowicz** [reply]
Currently, it is not possible
